src/main/java/core/basesyntax/impl/StorageImpl.java

```
@@ -3,17 +3,52 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int NOT_EXISTS = -1;
+    private static final int MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int currentSize;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+        currentSize = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int isAvailablePosition = availableKey(key);
+        if (isAvailablePosition != NOT_EXISTS) {
+            values[isAvailablePosition] = value;
+            return;
+        }
+        if (currentSize < MAX_SIZE) {
```

it seems that we would just ignore the input if we have reached the max size, let's throw an exception in that case, to let the user of our class know that it did